### PR TITLE
Add migration to check if s3_data_key is null

### PR DESCRIPTION
**Pull request by Kristian-A**
# Description
Add a check if the `s3_plot_data` is NULL. For example markerHeatmapPlotMain is a plot in plots and tables, so it doesn’t have plotData object and its `s3_plot_data` will be NULL.

Nested deletion is implemented by taking everything prefixed by bucket/key. In case key (i.e. s3_plot_data) is NULL the elements staged for deletion will be everything prefixed by bucket which results in a trigger of a mass delete of everything in the bucket.

# Details
#### URL to issue
https://github.com/biomage-org/issues/issues/153

#### Link to staging deployment URL (or set N/A)
N/A

#### Integration test branch
master

# Merge checklist
Your changes will be ready for merging after all of the steps below have been completed.

### Code updates
Have best practices and ongoing refactors being observed in this PR
- [x] Migrated any selector / reducer used to the new format.
- [x] All new dependency licenses have been checked for compatibility

### Manual/unit testing
- [ ] Tested changes using InfraMock locally **or** no tests required for change, e.g. Kubernetes chart updates.
- [ ] Validated that current unit tests for code work as expected and are sufficient for code coverage **or** no unit tests required for change, e.g. documentation update.
- [ ] Unit tests written **or** no unit tests required for change, e.g. documentation update.

### Integration testing
**You must check the box below** to run integration tests on the latest commit on your PR branch.
Integration tests have to pass before the PR can be merged. Without checking the box, your PR
**will not pass** the required status checks for merging.

- [ ] Started end-to-end tests on the latest commit.

### Documentation updates
- [ ] Relevant Github READMEs updated **or** no GitHub README updates required.
- [ ] Relevant Wiki pages created/updated **or** no Wiki updates required.

### Optional
- [ ] Staging environment is unstaged before merging.
- [ ] Photo of a cute animal attached to this PR.

### CODE REVIEW

**Comment by Kristian-A** [comment]
This PR is ready for merge, but it requires a migration. In general is it okay to merge and run migrations (in staging) in the HMS repo @alexvpickering? Do we even have permissions to execute the workflows?

**Comment by alexvpickering** [comment]
> This PR is ready for merge, but it requires a migration. In general is it okay to merge and run migrations (in staging) in the HMS repo @alexvpickering? Do we even have permissions to execute the workflows?

In general it is okay to merge to master after review from someone (doesn't need to be me). Please let me know when a migration is needed, what type (e.g. sql), and if there is any check needed to make sure it went okay.

File: src/sql/migrations/20230725125625_check_delete_plot_s3path_isnull.js [hunk 1]
```diff
@@ -0,0 +1,93 @@
+const createTriggerDeleteLambda = (dbEnv, key, bucketName) => {
+  const triggerLambdaARN = `arn:aws:lambda:${process.env.AWS_REGION}:${process.env.AWS_ACCOUNT_ID}:function:delete-s3-file-lambda-${dbEnv}`;
+
+  // Removing the environment and account id from the bucket name.
+  // When making a migration, the environment would be development,
+  // due to the fact that the migration is ran locally,
+  // so we need to add the environment and accountID in the lambda itself
+  // const rawBucketName = bucketName.split('-').slice(0, -2).join('-');
```
Comment (alexvpickering) [inline src/sql/migrations/20230725125625_check_delete_plot_s3path_isnull.js:8]: remove this comment

Comment (alexvpickering) [inline src/sql/migrations/20230725125625_check_delete_plot_s3path_isnull.js:8]:
```suggestion
```

**Review by alexvpickering — APPROVED** [review]

**Pull request merged**